wsmaster/codenvy-hosted-api-resource/src/main/java/com/codenvy/resource/api/ResourceAggregator.java

```diff
@@ -105,7 +104,61 @@ public ResourceAggregator(Set<ResourceType> resourcesTypes) {
         }
 
         if (!missingResources.isEmpty()) {
-            throw new NoEnoughResourcesException(totalResources, resourcesToDeduct, missingResources);
+            throw new NoEnoughResourcesException(sourceResources, resourcesToDeduct, missingResources);
+        }
+
+        return new ArrayList<>(result.values());
+    }
+
+    /**
+     * Returns list which contains resources from specified {@code sourceResources}
+     * which have excessive amount in compare to specified {@code resourcesToCompare}.
+     *
+     * <p>Example :
+     * <pre>
+     * |      \      | Source    | To compare| Result   |
+     * |:------------|:----------|:----------|:---------|
+     * | Resource1   | 5         | 3         | 2        |
+     * | ----------- | --------- | --------- | -------- |
+     * | Resource2   | -         | 9         | -        |
+     * | ----------- | --------- | --------- | -------- |
+     * | Resource3   | 1         | -         | 1        |
+     * | ----------- | --------- | --------- | -------- |
+     * </pre>
+     *
+     * @param sourceResources
+     *         the source resources
+     * @param resourcesToCompare
+     *         the resources which should be compared to {@code sourceResources}
+     * @throws IllegalStateException
```

Comment:
Good catch. I'll change throwing of IllegalStateException to IllegalArgumentException